ex5/ex5.c

```diff
@@ -15,7 +15,36 @@ char* msg3 = "hello world #3";
 
 int main()
 {
-    // Your code here
+    char inbuf[MSGSIZE];
+    int p[2];
+
+    if (pipe(p) < 0) 
+    {
+        fprintf(stderr, "Nobody here but us chickens...\n");
+        exit(1);
+    }
+
+    int rc = fork(); 
+    if (rc < 0) 
+    {
+      fprintf(stderr, "Nobody here but us chickens...\n");
+      exit(1);
+    }
+    else if (rc == 0)
+    {
+        write(p[1], msg1, MSGSIZE);
+        write(p[1], msg2, MSGSIZE);
+        write(p[1], msg3, MSGSIZE);
+    }
+    else
+    {
+        int wc = waitpid(rc, NULL, 0);
+        close(p[1]);
+        while(read(p[0], inbuf, MSGSIZE) > 0) 
```

Comment:
Pretty sure you're missing a `(` here in the `while`. Otherwise, your code is slightly more streamlined with just putting the `read` call inside the `while` instead of assigning it to a variable 👍